app/models/queen.rb

```diff
@@ -1,2 +1,7 @@
 class Queen < Piece
-end
\ No newline at end of file
+  def valid_move?(new_rank, new_file)
+    return false if is_move_malformed?(new_rank, new_file) || is_move_obstructed?(new_rank, new_file)
+    
+    rank == new_rank || file == new_file || (rank - new_rank).abs == (file - new_file).abs
```

Comment:
I would suggest reworking this slightly so that `is_move_obstructed?` only gets called at the end if needed. It's the most expensive step in the process (it needs to hit the DB and will usually loop a number of times), so that should make `Queen#valid_move?` more efficient.